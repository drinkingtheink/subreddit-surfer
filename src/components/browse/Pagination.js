import React, { Component } from "react"
import PropTypes from 'prop-types'

class Pagination extends Component {
  render() {
    const groupsPassed = this.props.groups && this.props.groups.length > 0
    const groupsCount = groupsPassed ? this.props.groups.length : 0

    return (
      <section className="search-pagination">
        <p className="pagination-meta">
          <strong>
            Page <span className="active-page-index">{this.props.activeResultsPageIndex + 1}</span> of <span className="total-groups-count">{groupsCount}</span>
          </strong>

          { groupsCount > 1 
            ? <span className="clickable-nav">
                {this.props.groups.map((result, i) => (
                  <span 
                    className={`group-click ${ i === this.props.activeResultsPageIndex ? 'active' : ''}`}
                    key={i}>
                  </span>
                ))}
              </span>
            : null
          }
        </p>
      </section>
    );
  }
}

Pagination.propTypes = {
  groups: PropTypes.array,
  activeResultsPageIndex: PropTypes.number,
  activeResultsGroup: PropTypes.object
}

export default Pagination;
