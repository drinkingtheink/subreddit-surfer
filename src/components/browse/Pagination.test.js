import React from 'react'
import { shallow } from 'enzyme'
import Pagination from './Pagination'
import { examplePaginationPayload } from './test/examplePaginationPayload'

const defaultComponent = shallow(<Pagination groups={ examplePaginationPayload.groups } activeResultsPageIndex={ examplePaginationPayload.activeResultsPageIndex } activeResultsGroup={ examplePaginationPayload.activeResultsGroup } />)

describe(`Pagination =====================`, () => {

  it(`renders without crashing`, () => {
    const component = shallow(<Pagination />)
    expect(component.exists()).toBe(true)
  });

  it(`renders with props`, () => {
    expect(defaultComponent).toMatchSnapshot();
  });

  it('displays the correct active page', () => {
    expect(defaultComponent.find('.active-page-index').text()).toEqual((examplePaginationPayload.activeResultsPageIndex + 1).toString())
  });

  it('displays the correct total groups count', () => {
    expect(defaultComponent.find('.total-groups-count').text()).toEqual((examplePaginationPayload.groups.length).toString())
  });

  it('displays the navigation if there are groups of search results', () => {
    expect(defaultComponent.find('.clickable-nav').exists()).toEqual(true)
  });

  it('does not display the navigation if there are no groups of search results', () => {
    const customWrapper = shallow(<Pagination groups={ [] } activeResultsPageIndex={ examplePaginationPayload.activeResultsPageIndex } activeResultsGroup={ examplePaginationPayload.activeResultsGroup } />)
    expect(customWrapper.find('.clickable-nav').exists()).toEqual(false)
  });
});