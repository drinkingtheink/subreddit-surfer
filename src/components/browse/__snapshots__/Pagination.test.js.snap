// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination ===================== renders with props 1`] = `ShallowWrapper {}`;
